lib/OpenQA/WebAPI/Controller/API/V1/Job.pm

```diff
@@ -597,9 +613,28 @@ interested in the status.
 =cut
 
 sub get_status ($self) {
+    my $follow = $self->param('follow');    # follow job clones and report most recent result for given id
     my @fields = qw(id state result blocked_by_id);
-    return unless my $job = $self->find_job_or_render_not_found($self->stash('jobid'));
-    $self->render(json => {map { $_ => $job->$_ } @fields});
+    my $query_jobid = $self->stash('jobid');
+    my @cloned_as;
+    my $job;
+    while (1) {
+        return unless $job = $self->find_job_or_render_not_found($query_jobid);
+        if ($follow && $job->clone_id) {
+            push(@cloned_as, $job->clone_id);
+            $query_jobid = $job->clone_id;
+            next;
+        }
+        last;
+    }
+    my $json = {map { $_ => $job->$_ } @fields};
+    if ($query_jobid != $self->stash('jobid')) {
+        # if follow=1 return cloned job with original id
+        # but indicate clone chain in cloned_as
+        $json->{id} = $self->stash('jobid');
```

Comment:
This shouldn't return the original job ID anymore, or should it? I would say that with `--follow` one expects all data to be returned for the latest job including the ID. One can still read `cloned_as` to get the original ID.

Reply:
If it doesn't, it won't work for the API methods for `openqa-cli monitoring` mentioned explicitly in the ticket. And there would be no way for the user calling openqa-cli to know which job id corresponds to which id they passed as a parameter.